modules/geolocation/geolocation.py

```diff
@@ -129,3 +130,61 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+
+    def get_best_location(self,inputLocationTupleList):
+
+        # Inner helper function that checks if the trimmed array is empty ie all values were too far apart from the median
+        # Returns median if array is empty, otherwise finds the average of the trimmed array
+        def check_for_empty(trimmedArray,median):
```

Comment:
Move this outside of the get_best_location() function, add two underscores: `__check_for_empty()` to indicate that it is private.

Also, a better name for the function may be: `__get_average_otherwise_median()`